### ISSUE TITLE
Pretrained inference bug across pytorch versions

### ISSUE THREAD

**Issue by alejodosr** [pytorch/pytorch#14721]
## 🐛 Bug
Inference issue with Resnet50 pretrained model (different images evaluation)
Not sure if it is a bug, but different behavior with same code (but different pytorch versions)

Here, context of the problem:

# Common code:

```
from torchvision import datasets, models, transforms
import torchvision
from PIL import Image
from torch.autograd import Variable

normalize = transforms.Normalize(
   mean=[0.485, 0.456, 0.406],
   std=[0.229, 0.224, 0.225]
)
preprocess = transforms.Compose([
   transforms.Resize(256),
   transforms.CenterCrop(224),
   transforms.ToTensor(),
   normalize
   # transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
])

resnet_model = torchvision.models.resnet50(pretrained=True, num_classes=1000)
resnet_model.eval()

for i in range(4):
    # An instance of your model.
    img_pil = Image.open("/home/alejandro/workspace/uav_detection/images/" + str(i + 1) + ".jpg")
    # img_pil.show()
    img_tensor = preprocess(img_pil).float()
    img_tensor = img_tensor.unsqueeze_(0)

    fc_out = resnet_model(Variable(img_tensor))

    output = fc_out.detach().numpy()
    print(output.argmax())
```
# ENVIRONMENT 1 (Pytorch built from source):
PyTorch version: 1.0.0a0+37627a1
Is debug build: No
CUDA used to build PyTorch: 9.2.148

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: version 3.12.2

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 950M
Nvidia driver version: 396.54
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a
/usr/local/MATLAB/MATLAB_Runtime/v94/bin/glnxa64/libcudnn.so.7.0.3
/usr/local/MATLAB/R2018a/bin/glnxa64/libcudnn.so.7.0.3
/usr/local/cuda-9.2/lib64/libcudnn.so
/usr/local/cuda-9.2/lib64/libcudnn.so.7
/usr/local/cuda-9.2/lib64/libcudnn.so.7.4.1
/usr/local/cuda-9.2/lib64/libcudnn_static.a
/usr/local/cuda-9.2/lib64/libcudnn_static_v7.a

Versions of relevant libraries:
[pip] Could not collect
[conda] magma-cuda92              2.4.0                         1    pytorch
[conda] torch                     1.0.0a0+37627a1           <pip>
[conda] torchvision               0.2.1                     <pip>

RESULTS:
```
918
918
918
918
```
# ENVIRONMENT 2:
PyTorch version: 0.4.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
CMake version: version 3.12.2

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 950M
Nvidia driver version: 396.54
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.1
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a
/usr/local/MATLAB/MATLAB_Runtime/v94/bin/glnxa64/libcudnn.so.7.0.3
/usr/local/MATLAB/R2018a/bin/glnxa64/libcudnn.so.7.0.3
/usr/local/cuda-9.2/lib64/libcudnn.so
/usr/local/cuda-9.2/lib64/libcudnn.so.7
/usr/local/cuda-9.2/lib64/libcudnn.so.7.4.1
/usr/local/cuda-9.2/lib64/libcudnn_static.a
/usr/local/cuda-9.2/lib64/libcudnn_static_v7.a

Versions of relevant libraries:
[pip] Could not collect
[conda] pytorch                   0.4.1           py27__9.0.176_7.1.2_2    pytorch
[conda] torchfile                 0.1.0                      py_0    conda-forge
[conda] torchvision               0.2.1                    py27_1    pytorch

RESULTS:
```
834
208
285
478
```
# Additional notes
Same constant results (918) when tracing the model and loading it with the C++ API.

**Comment by ptrblck** [comment]
Could reproduce this issue in `37627a1`. In `1.0.0a0+4c11dee` it looks like it's working fine.
Some debugging and more info can be found [here](https://discuss.pytorch.org/t/how-to-classify-single-image-using-loaded-net/1411/28?u=ptrblck).

**Comment by soumith** [comment]
@ptrblck it might be fixed by https://github.com/pytorch/pytorch/pull/13274 or https://github.com/pytorch/pytorch/pull/13211 maybe?

**Comment by soumith** [comment]
i'm closing the issue as it's fixed in master and 1.0.0
